src/Microsoft.Data.SqlClient/add-ons/AzureKeyVaultProvider/AzureSqlKeyCryptographer.cs

```diff
@@ -208,10 +216,13 @@ private void CreateKeyClient(Uri vaultUri)
         /// <param name="masterKeyVersion">The version of the key</param>
         private void ParseAKVPath(string masterKeyPath, out Uri vaultUri, out string masterKeyName, out string masterKeyVersion)
         {
-            Uri masterKeyPathUri = new Uri(masterKeyPath);
+            Uri masterKeyPathUri = new(masterKeyPath);
             vaultUri = new Uri(masterKeyPathUri.GetLeftPart(UriPartial.Authority));
             masterKeyName = masterKeyPathUri.Segments[2];
             masterKeyVersion = masterKeyPathUri.Segments.Length > 3 ? masterKeyPathUri.Segments[3] : null;
+
+            AKVEventSource.Log.TryTraceEvent("Received Key Name: {0}", masterKeyName);
+            AKVEventSource.Log.TryTraceEvent("Received Key Version: {0}", masterKeyVersion);
```

Comment:
Any security concerns in logging the entire version?

Reply:
All parameters are split at 10 character length universally:
https://github.com/dotnet/SqlClient/pull/1174/files#diff-fdef6a270d141d4cf0ddd2de0debdba3e6c66dc8f99fbc072dcbf3a8b770f59bR76